engine/graphics/src/vulkan/graphics_vulkan_device.cpp

```diff
@@ -247,40 +247,6 @@ namespace dmGraphics
         }
     }
 
-    VkResult OneTimeCommandBuffer::Begin()
-    {
-        assert(m_Context != 0);
-
-        VkCommandBuffer vk_command_buffer;
-        CreateCommandBuffers(m_Context->m_LogicalDevice.m_Device, m_Context->m_LogicalDevice.m_CommandPool, 1, &vk_command_buffer);
-        VkCommandBufferBeginInfo vk_command_buffer_begin_info;
-        memset(&vk_command_buffer_begin_info, 0, sizeof(VkCommandBufferBeginInfo));
-
-        vk_command_buffer_begin_info.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
-        vk_command_buffer_begin_info.flags = VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT;
-        VkResult res = vkBeginCommandBuffer(vk_command_buffer, &vk_command_buffer_begin_info);
-        m_CmdBuffer = vk_command_buffer;
-        return res;
-    }
-
-    VkResult OneTimeCommandBuffer::End()
-    {
-        vkEndCommandBuffer(m_CmdBuffer);
-
-        VkSubmitInfo vk_submit_info = {};
-        vk_submit_info.sType              = VK_STRUCTURE_TYPE_SUBMIT_INFO;
-        vk_submit_info.commandBufferCount = 1;
-        vk_submit_info.pCommandBuffers    = &m_CmdBuffer;
-
-        VkResult res = vkQueueSubmit(m_Context->m_LogicalDevice.m_GraphicsQueue, 1, &vk_submit_info, VK_NULL_HANDLE);
-        vkQueueWaitIdle(m_Context->m_LogicalDevice.m_GraphicsQueue);
-        vkFreeCommandBuffers(m_Context->m_LogicalDevice.m_Device, m_Context->m_LogicalDevice.m_CommandPool, 1, &m_CmdBuffer);
-
-        m_Context   = 0;
-        m_CmdBuffer = VK_NULL_HANDLE;
-        return res;
-    }
```

Comment:
Replaced and/or removed. We don't do these hard syncronizations any longer (vkQueueWaitIdle basically stalls the entire GPU).